### PR TITLE
tsify more addons modules

### PR DESCRIPTION
**Pull request by MayhemYDG**

### CODE REVIEW

File: packages/react-data-grid-addons/src/formatters/ImageFormatter.tsx [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
+
+interface Props {
+  /** image url, used as background-image */
+  value: string;
+}
+
+export default function ImageFormatter({ value }: Props) {
+  return <div className="react-grid-image" style={{ backgroundImage: `url(${value})` }} />;
```
Comment (MayhemYDG) [inline packages/react-data-grid-addons/src/formatters/ImageFormatter.tsx]: The previous `ImageFormatter` implementation loaded the url in the background before applying the style, I don't know why, doesn't seem worthwhile.

File: packages/react-data-grid-addons/src/menu/ContextMenu.js [hunk 2]
```diff
@@ -1,18 +0,0 @@
...
-import { ContextMenu } from 'react-contextmenu';
-
-export default class ReactDataGridContextMenu extends React.Component {
-  static propTypes = {
-    children: PropTypes.node
-  };
-
-  render() {
-    return (
-      <ContextMenu identifier="reactDataGridContextMenu">
```
Comment (MayhemYDG) [inline packages/react-data-grid-addons/src/menu/ContextMenu.js]: Bug uncovered: `identifier` is not a valid prop, `id` is though.

File: packages/react-data-grid-addons/src/cells/headerCells/filters/NumericFilter.tsx [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+  /** Validates the input */
+  function handleKeyPress(event: React.KeyboardEvent<HTMLInputElement>) {
+    const result = /[><,0-9-]/.test(event.key);
+    if (result === false) {
+      event.preventDefault();
+    }
+  }
+
+  function handleChange(event: React.ChangeEvent<HTMLInputElement>) {
+    const value = event.target.value;
```
Comment (amanmahajan7) [inline packages/react-data-grid-addons/src/cells/headerCells/filters/NumericFilter.tsx]: `const { value } = e.target`

File: packages/react-data-grid-addons/src/cells/headerCells/filters/NumericFilter.tsx [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+    cursor: 'help'
+  };
+
+  const tooltipText = 'Input Methods: Range (x-y), Greater Then (>x), Less Then (<y)';
+
+  return (
+    <div>
+      <div style={columnStyle}>
+        <input
+          key={inputKey}
```
Comment (amanmahajan7) [inline packages/react-data-grid-addons/src/cells/headerCells/filters/NumericFilter.tsx]: missing `type="text"`

File: packages/react-data-grid-addons/src/cells/headerCells/filters/NumericFilter.tsx [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+
+
+function filterValues(row: RowData, columnFilter: { filterTerm: { [key in string]: Rule } }, columnKey: string) {
+  if (columnFilter.filterTerm == null) {
+    return true;
+  }
+
+  // implement default filter logic
+  const value = parseInt(row[columnKey] as string, 10);
+  for (const ruleKey in columnFilter.filterTerm) {
```
Comment (amanmahajan7) [inline packages/react-data-grid-addons/src/cells/headerCells/filters/NumericFilter.tsx]: `for (const rule of columnFilter.filterTerm)`

File: packages/react-data-grid-addons/src/toolbars/Toolbar.tsx [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+            {props.filterRowsButtonText}
+          </button>
+        )}
+        {props.children}
+      </div>
+    </div>
+  );
+}
+
+Toolbar.defaultProps = {
```
Comment (amanmahajan7) [inline packages/react-data-grid-addons/src/toolbars/Toolbar.tsx]: > For stateless function components (SFCs) use ES2015 default initializers for SFCs:

https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-0.html

File: packages/react-data-grid-addons/src/cells/headerCells/filters/NumericFilter.tsx (hunk 5 above)
Comment (MayhemYDG) [reply packages/react-data-grid-addons/src/cells/headerCells/filters/NumericFilter.tsx]: It's not an iterable.

File: packages/react-data-grid-addons/src/cells/headerCells/filters/NumericFilter.tsx (hunk 4 above)
Comment (MayhemYDG) [reply packages/react-data-grid-addons/src/cells/headerCells/filters/NumericFilter.tsx]: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input

>If this attribute is not specified, the default type adopted is `text`.

I can add it, but it's not necessary. wdyt?

File: packages/react-data-grid-addons/src/cells/headerCells/filters/NumericFilter.tsx (hunk 5 above)
Comment (amanmahajan7) [reply packages/react-data-grid-addons/src/cells/headerCells/filters/NumericFilter.tsx]: Ah, I got confused with `filterTerm: Rule[] | null;`

File: packages/react-data-grid-addons/src/cells/headerCells/filters/NumericFilter.tsx (hunk 4 above)
Comment (amanmahajan7) [reply packages/react-data-grid-addons/src/cells/headerCells/filters/NumericFilter.tsx]: I usually prefer to add it to be explicit but it is fine in this case

**Pull request merged**